projects/plugins/jetpack/extensions/blocks/contact-info/contact-info.php

```diff
@@ -9,34 +9,40 @@
 
 use Automattic\Jetpack\Blocks;
 
-Blocks::jetpack_register_block(
-	__DIR__,
-	array(
-		'render_callback' => array( 'Jetpack_Contact_Info_Block', 'render' ),
-	)
-);
+add_action(
+	'init',
+	function () {
+		Blocks::jetpack_register_block(
+			__DIR__,
+			array(
+				'render_callback' => array( 'Jetpack_Contact_Info_Block', 'render' ),
+			)
+		);
 
-Blocks::jetpack_register_block(
-	'jetpack/address',
-	array(
-		'parent'          => array( 'jetpack/contact-info' ),
-		'render_callback' => array( 'Jetpack_Contact_Info_Block', 'render_address' ),
-	)
-);
+		Blocks::jetpack_register_block(
+			'jetpack/address',
+			array(
+				'parent'          => array( 'jetpack/contact-info' ),
+				'render_callback' => array( 'Jetpack_Contact_Info_Block', 'render_address' ),
+			)
+		);
 
-Blocks::jetpack_register_block(
-	'jetpack/email',
-	array(
-		'parent'          => array( 'jetpack/contact-info' ),
-		'render_callback' => array( 'Jetpack_Contact_Info_Block', 'render_email' ),
-	)
-);
+		Blocks::jetpack_register_block(
+			'jetpack/email',
+			array(
+				'parent'          => array( 'jetpack/contact-info' ),
+				'render_callback' => array( 'Jetpack_Contact_Info_Block', 'render_email' ),
+			)
+		);
 
-Blocks::jetpack_register_block(
-	'jetpack/phone',
-	array(
-		'parent'          => array( 'jetpack/contact-info' ),
-		'render_callback' => array( 'Jetpack_Contact_Info_Block', 'render_phone' ),
-	)
+		Blocks::jetpack_register_block(
+			'jetpack/phone',
+			array(
+				'parent'          => array( 'jetpack/contact-info' ),
+				'render_callback' => array( 'Jetpack_Contact_Info_Block', 'render_phone' ),
+			)
+		);
+	}
```

Comment:
Should we move all this to a `Jetpack_Contact_Info_Block::register_block()` function?

Reply:
We could. I followed the style in [recipe.php](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/extensions/blocks/recipe/recipe.php). But I could instead add the hook in the main file, and the functionality in `Jetpack_Contact_Info_Block`. 

Alternatively we could do what other simpler blocks do, like [Calendly](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/extensions/blocks/calendly/calendly.php) and Button and have everything in one file.

It seems a little out of scope of this PR to do that, but since we're modifying the file anyway I decided to make the (latter) change.

Reply:
I think recipe.php is another odd one out. 🙂 Most seem to use a `register_block` function in the namespace, but since we have a class we may as well use it.

Personally I'd probably stick with the class rather than the bunch-of-functions-in-a-namespace, but that's probably because of my background with MediaWiki where everything was auto-loaded (and functions can't be autoloaded) versus Jetpack and WordPress where most things are manually `require`d. I won't object to 053e9b8 if that's the way you want to go.

Reply:
With a class, it looks like we'd need the two files (class and non-class), then require the class file from the virtually empty non-class file?
Either way, I can create another issue to follow up with, to look at both Contact Info and Recipe to make sure they're both consistent.

Reply:
That's right. The non-class file would require the class file and call `add_action( 'init', array( Jetpack_Contact_Info_Block::class, 'register_block' ) );` or the like.